feats/storage.py

```diff
@@ -11,9 +13,28 @@
 If this Storage has not yet seen a key, it should return an empty sequence.
 """
 
-Memory: Storage = lambda: defaultdict(list)
+Memory: Storage = lambda: defaultdict(MemoryList)
 """
 The Memory storage keeps all data in memory. When the application exits, all data
 will be lost.
 Mainly useful for testing environments.
 """
+class MemoryList(list):
```

Comment:
What about something which describes the properties of this list, namely that it is append only & the values stored are immutable?